footil/date.py

```diff
@@ -8,9 +8,10 @@
 
 
 def _modify_date(
-    dt: str,
+    dt: str or datetime.date,
     fmt: str,
     modify_fun: callable,
+    res_type: Optional[str] = 'str',
```

Comment:
there are only two possible outcomes, so string value as an option is too broad. Better to just use boolean, indicating to convert to date or string. if False, do the opposite.

On the other hand, it looks like this method is trying to do too much. There are too many args already. We should probably leave this method as is and then write wrapper, that can accept date object and convert to another date object. More rarely I think, it will be the case, when we need to for example give date obj and return str or vise versa. And if such thing happens, I think it will be no problem to convert separately.

Need to refactor in a way that both helpers would use same lower level functions, but with different types. date obj helper, won't need format arguments, so it should be even simpler and in our Odoo case that would be used more than current implementation.